src/main.rs

```diff
@@ -31,24 +33,66 @@ fn send_usb_request(device: &Device) {
                 println!("HUION_PAD_MODE={s}");
             }
         });
+
+    Ok(())
 }
 
-fn main() {
+fn send_usb() -> Result<()> {
     let ctx = Context::new().unwrap();
 
     const HUION_VENDOR_ID: u16 = 0x256C;
 
-    ctx.devices()
-        .unwrap()
-        .iter()
-        .filter(|d| {
-            if let Ok(desc) = d.device_descriptor() {
-                desc.vendor_id() == HUION_VENDOR_ID
-            } else {
-                false
+    for device in ctx.devices().unwrap().iter().filter(|d| {
```

Comment:
Can `for_each` be used here instead of a for loop? It'd follow a bit better the functional style.

Reply:
Fixed, has to be `try_for_each()` since we want to return a result.